ttnn/cpp/ttnn/operations/matmul/device/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp

```diff
@@ -272,8 +272,12 @@ void kernel_main() {
             noc_async_write_multicast(
                 in1_start_address, in1_multicast_data_addr, in1_block_size_bytes, in1_mcast_num_cores, true, true);
 
-            // Note: no need for write barrier, since these two multicasts are done on the same noc id, same vc, same
-            // cmd_buf Also, this only works because we are setting VCs statically (using NOC_CMD_STATIC_VC).
+            // Note: no need for write barrier, since these two multicasts are done on the same noc id and same vc even though cmd bufs are different
+            // Also, this only works because we are setting VCs statically (using NOC_CMD_STATIC_VC).
+#ifdef ARCH_BLACKHOLE
+            // On Blackhole the flush is needed because the commands go into separate cmd buffer FIFOs and may not be sent in order they are issued
+            noc_async_writes_flushed();
+#endif
```

Comment:
should this flush after semaphore call ?

Reply:
it seems the flush is needed before the sem call